invenio_oauthclient/contrib/cern.py

```diff
@@ -210,13 +216,5 @@ def account_setup(remote, token):
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
-        current_user.modified = True
-        current_user.save()
-
-        if user and not any([user.family_name, user.given_names]):
-            user.family_name = res['Lastname'][0]
-            user.given_names = res['Firstname'][0]
-
-            db.session.add(user)
-            current_user.reload()
+        res['groups'] = fetch_groups(res['Group'])
```

Comment:
Is this really needed? Why is necessary to parse the groups and remove some of them?